oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java

```diff
@@ -107,15 +149,51 @@ public void addHandler(Object handler, List<HttpMethod> httpMethods) {
             "Bind handler {} into http server {}:{}",
             handler.getClass().getSimpleName(), config.getHost(), config.getPort()
         );
-
+        if (config.isEnableTLS()) {
+            handlers.add(handler);
+        }
         sb.annotatedService()
           .pathPrefix(config.getContextPath())
           .build(handler);
         this.allowedMethods.addAll(httpMethods);
     }
 
+    public void updateCert() {
+        FileTime lastModifiedTimeCertNow;
+        FileTime lastModifiedTimeKeyNow;
+        try {
+            lastModifiedTimeCertNow = Files.getLastModifiedTime(tlsCertChainPath);
+            lastModifiedTimeKeyNow = Files.getLastModifiedTime(tlsKeyPath);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        if (lastModifiedTimeCertNow.equals(lastModifiedTimeCert)
+                && lastModifiedTimeKeyNow.equals(lastModifiedTimeKey)) {
+            return;
+        }
+        log.info("TLS cert chain file or TLS key file has been updated, reloading...");
+        httpServer.reconfigure(sb -> {
+            try (InputStream cert = new FileInputStream(tlsCertChainPath.toFile());
+                 InputStream key = PrivateKeyUtil.loadDecryptionKey(tlsKeyPath.toString())) {
+                sb.tls(cert, key);
+            } catch (IOException e) {
+                throw new IllegalArgumentException(e);
+            }
+            normalInitialize(sb);
+            sb.annotatedService()
+              .pathPrefix(config.getContextPath())
+              .build(handlers.toArray());
```

Comment:
Can you confirm this is needed when we only want to reconfigure the tls cert? I think Armeria will maintain the existing service endpoint if we didn't add new or remove existing service.

Reply:
`reconfigure` method will creat a new ServerBuild. So I think it is needed.
![image](https://user-images.githubusercontent.com/74546965/214071296-4489a8df-caab-4d82-afbc-bb2ef577edbb.png)
![image](https://user-images.githubusercontent.com/74546965/214071354-9cd61611-7bb6-4681-a5fb-dc4033673b7e.png)

Reply:
`sb.buildServerConfig(config())` only set the host and port.

[unit test of Armeria](https://github.com/line/armeria/blob/871d87297e4d051241589cb1ae95641cbc83f880/core/src/test/java/com/linecorp/armeria/client/ReconfigurableServerTest.java#L120) can also prove it.